backend/curriculum_tracking/tests/test_AgileCard.py

```diff
@@ -688,3 +694,199 @@ def test_cancel_workshop_attendance(self):
         self.assertIsNone(self.card.workshop_attendance)
         self.assertEqual(WorkshopAttendance.objects.count(), 0)
         self.assertEqual(self.card.status, AgileCard.READY)
+
+
+class TestingForTheStatsAPI(TestCase):
+
+    def test_creating_cards_and_testing_functionality_of_stats_api(self):
+
+        # One card with 4 reviews to follow on it
+        self.card_1 = AgileCardFactory(
+            status=AgileCard.IN_PROGRESS,
+        )
+
+        self.project_1 = self.card_1.recruit_project
```

Comment:
If a variable is only used inside one function then there is no reason to use self. In fact using `self` when you shouldnt is bad practice.

Imagine if you have a complicated class that has 50 functions. Inside each function, whenever there is a variable declared then it is added to the instance `self.thingy = whatever`.

Now since there are 50 functions some names get reused in different functions. So calling one function can ffect how other functions behave in unintended ways. This would lead to bugs.

It would also lead to confusion - imagine a new developer joining the team and needing to understand the code. There is no way of seeing which of the variables are only meant to be in scope inside the function or which variables are meant to be accessed on the instance. This becomes like spegetti which leads to wasted time and more bugs.

Only use self if: You need the variable to be in scope outside the current function.

You might want to look into the principle of least privilage

Reply:
I removed most usages of 'self' outside the setUp function.